### PR TITLE
bpo-34604: Fix possible mojibake in `pwd.getpwnam` and `grp.getgrnam`

### PR DESCRIPTION
**Pull request by william-gr**
Spotted by:	@vstinner

https://bugs.python.org/issue34604

### CODE REVIEW

**Comment by william-gr** [comment]
@vstinner here is the follow up PR as discussed in https://github.com/python/cpython/pull/7081

Please let me know if thats what you were meaning.

Thanks!

**Comment by william-gr** [comment]
The build failed probably because I change pwd.getpwnam signature from `name: unicode` to `arg: unicode`, however I am unsure how to fix that. Investigating.

File: Misc/NEWS.d/next/Library/2018-09-07-10-16-34.bpo-34604.xL7-kG.rst [hunk 1]
```diff
@@ -0,0 +1,2 @@
+Fix possible mojibake in `pwd.getpwnam` and `grp.getgrnam`. Patch by William
```
Comment (vstinner) [inline Misc/NEWS.d/next/Library/2018-09-07-10-16-34.bpo-34604.xL7-kG.rst]: would you mind to add "... in the error message." to clarify that it doesn't affect the regular case?

**Comment by william-gr** [comment]
@vstinner sure, updated.

Thanks

**Comment by vstinner** [comment]
@william-gr: it seems like the the bot failed to backport the fix. Would you mind to try to create 3.6 and 3.7 backport pull requests?

**Pull request merged**